Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -45,44 +59,96 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
 	if (ensureMsgf(LocalPC, TEXT("ASSERT: [%i] %hs:\n'LocalPC' is null!"), __LINE__, __FUNCTION__))
 	{
 		LocalPC->OnAnyCinematicStarted.AddDynamic(this, &APSStarActor::OnAnyCinematicStarted);
+		Character->OnPlayerTypeChanged.AddUniqueDynamic(this, &ThisClass::OnPlayerTypeChanged);
 	}
 }
 
+// Called when the current  player was changed
+void APSStarActor::OnPlayerTypeChanged(FPlayerTag PlayerTag)
+{
+	InitStarMenuAnimation();
+}
+
+// Called when the current game state was changed
+void APSStarActor::OnGameStateChanged(ECurrentGameState GameState)
+{
+	InitStarMenuAnimation();
+}
+
 // Is called when any cinematic started
 void APSStarActor::OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator)
+{
+	SetAnimationStartTime(StartTimeHideStarsInternal);
+	TryPlayHideStarAnimation();
+}
+
+// Is called to set a start time for animations
+void APSStarActor::SetAnimationStartTime(float& StartTime)
 {
 	const UWorld* World = GetWorld();
 	check(World);
 
-	StartTimeHideStarsInternal = World->GetTimeSeconds();
+	StartTime = World->GetTimeSeconds();
 
-	// Enable tick and process this frame
+	// Enable tick for the actor
 	SetActorTickEnabled(true);
-	TryPlayHideStarAnimation();
 }
 
-// Hiding stars with animation
-void APSStarActor::TryPlayHideStarAnimation()
+// Called to initialize the Star menu animation
+void APSStarActor::InitStarMenuAnimation()
 {
-	if (!StartTimeHideStarsInternal)
+	if (ensureMsgf(UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState(), TEXT("%s: 'GetMyGameState' is null"), *FString(__FUNCTION__))
+		&& UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState() == ECurrentGameState::Menu)
 	{
-		return;
+		SetAnimationStartTime(StartTimeMenuStarsInternal);
+		TryPlayMenuStarAnimation();
+	}
+	else
+	{
+		SetActorTickEnabled(false);
+		StartTimeMenuStarsInternal = 0.f;
 	}
+}
+
+// Hiding stars with animation in main menu when cinematic is start to play
+void APSStarActor::TryPlayHideStarAnimation()
+{
+	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
+	TryPlayStarAnimation(StartTimeHideStarsInternal, CurrentRow.HideStarsAnimation);
+}
 
+// Menu stars with animation in main menu idle 
+void APSStarActor::TryPlayMenuStarAnimation()
+{
 	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
-	if (!CurrentRow.HideStarsAnimation)
+	TryPlayStarAnimation(StartTimeMenuStarsInternal, CurrentRow.MenuStarsAnimation);
+}
+
+// Playing star animation
+void APSStarActor::TryPlayStarAnimation(float& StartTime, UCurveTable* AnimationCurveTable)
+{
+	if (!StartTime || !AnimationCurveTable)
 	{
 		SetActorTickEnabled(false);
-		StartTimeHideStarsInternal = 0.f;
+		StartTime = 0.f;
 		return;
 	}
 
-	const float SecondsSinceStart = GetWorld()->GetTimeSeconds() - StartTimeHideStarsInternal;
+	const float SecondsSinceStart = GetWorld()->GetTimeSeconds() - StartTime;
```

Comment:
I have compile error, shows World's header is not included. I have next settings in ProgressionSystem.build.cs:
```
		OptimizeCode = CodeOptimization.Never; //@TODO debug
		bUseUnity = false; //@TODO debug
		PCHUsage = PCHUsageMode.NoPCHs; //@TODO debug
		CppStandard = CppStandardVersion.Latest;
		bEnableNonInlinedGenCppWarnings = true;
		bUseUnity = false;
```

Please do:
```
#include "Engine/World.h"
```

Reply:
fixed https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/2fac0deafb94929a1c5026c9e0dec721ffe1c04c